File: training/vue/src/de/sots/cellarsandcentaurs/domain/model/Centaur.ts
import { Creature } from './Creature'
import { CreatureId } from './CreatureId'
import { CreatureType } from './CreatureType'
import { Speed } from './Speed'
import { SpeedType } from './SpeedType'
import { Roams } from './Roams'

@Roams('cellar')
export class Centaur extends Creature {
  private readonly XPValue = 100

  constructor(id: CreatureId, walkingSpeed: Speed) {
    super(id, CreatureType.MONSTROSITY)
    this.setSpeeds(new Map([[SpeedType.WALKING, walkingSpeed]]))
  }

  getXPValue(): number {
    return this.XPValue
  }
}

File: training/swift/Sources/CellarsAndCentaurs/Domain/Model/Centaur.swift
final class Centaur: Creature {
    let XPValue: Int

    init(id: CreatureId, XPValue: Int) {
        self.XPValue = XPValue
        super.init(id: id, type: .monstrosity)
        speeds[.walking] = Speed(50)
    }

    override func takeDamage(_ amount: Int) {
        super.takeDamage(amount)
        if hitPoints?.current == 0 {
            speeds[.walking] = Speed.stationary
        }
    }
}
